### PR TITLE
fix(router): fix parameter and query parameter types `any` -> `string | string []`

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: packages/router/src/router_state.ts [hunk 1]
```diff
@@ -353,12 +392,13 @@ export function advanceActivatedRoute(route: ActivatedRoute): void {
   }
 }
 
-
-export function equalParamsAndUrlSegments(
+// Recursively check that the active route hierarchy have the same path and parameter
+export function equalMatrixParamsAndUrlSegments(
```
Comment (vicb) [inline packages/router/src/router_state.ts]: revert name change

**Comment by vicb** [comment]
won't fix for now.

**Comment by ocombe** [comment]
@jasonaden as requested, I've rebased this PR onto master

**Review by jasonaden — APPROVED** [review]

**Comment by jasonaden** [comment]
Presubmit run: https://test.corp.google.com/ui#id=OCL:167582004:BASE:167582027:1504625128842:faff2e27

**Comment by jasonaden** [comment]
Closing this as it's not going to make it.

**Pull request closed without merging**